Mutagen.Bethesda.Analyzers.Skyrim/Util/VoiceTypeAssetLookupUtil.cs

```diff
@@ -0,0 +1,28 @@
+using Mutagen.Bethesda.Plugins.Assets;
+using Mutagen.Bethesda.Plugins.Cache;
+using Mutagen.Bethesda.Skyrim.Records.Assets.VoiceType;
+
+namespace Mutagen.Bethesda.Analyzers.Skyrim.Util;
+
+public static class VoiceTypeAssetLookupUtil
+{
+    private static readonly Lock Lock = new();
+    private static VoiceTypeAssetLookup? _voiceTypeAssetLookup;
+
+    public static VoiceTypeAssetLookup GetVoiceTypeAssetLookup(ILinkCache linkCache)
+    {
+        // TODO: Replace with something that is updated when the link cache is updated
```

Comment:
Can you talk me through what the concern is that requires a lock?
Is it because it might be a mutable cache passed in?   
I've been coming at analyzers from the angle that it'll be mostly all functional (eg, immutable)

Reply:
Because we are running analyzers in parallel and we might try to create multiple immutable caches at the same time when it is still null atm.